src/libs/actions/Report.js

```diff
@@ -293,8 +316,23 @@ function fetchChatReportsByIDs(chatList) {
             // personal details of all the participants and even link up their avatars to report icons.
             const reportIOUData = {};
             _.each(fetchedReports, (report) => {
-                const simplifiedReport = getSimplifiedReportObject(report);
-                simplifiedReports[`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`] = simplifiedReport;
+                // Do not store IOU reports as 'report_' Onyx keys.
+                if (lodashGet(report, 'reportNameValuePairs.type') === 'chat') {
+                    const simplifiedReport = getSimplifiedReportObject(report);
+                    simplifiedReports[`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`] = simplifiedReport;
+                } else if (lodashGet(report, 'reportNameValuePairs.type') === 'iou') {
+                    // If there's a new IOU report, find corresponding chat report and link them
+                    findChatReportID([report.ownerEmail, report.managerEmail]).then((chatReportID) => {
+                        const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${report.reportID}`;
+                        const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`;
+                        Onyx.merge(iouReportKey, getSimplifiedIOUReport(report, chatReportID));
+                        const chatReportObject = {};
+                        chatReportObject.iouReportID = report.reportID;
+                        chatReportObject.hasOutstandingIOU = report.stateNum === 1
+                            && report.total !== 0;
+                        Onyx.merge(reportKey, chatReportObject);
```

Comment:
Don't think we should be merging to Onyx within a loop here. Couldn't we follow the pattern from the first if condition, and collect all the reports in `reportIOUData` and then use mergeCollection on line 354 to merge once?

Reply:
Since we are using `findChatReportID` here and it returns a Promise. I don't think we can follow the same pattern here without refactoring the whole function as we cannot await promise synchronously. Is there a better way to handle this part without using `Promise` or `Onyx.merge`?

I think this would be OK because we call fetchChatReportsByIDs with multiple chat report IDs or a single IOU report ID. And this merge action is called only for that single IOU report ID.